benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -135,6 +135,11 @@ private synchronized void handleStartState(String testID) {
       // wait for task to complete
       future.get();
 
+      if (isTerminated(testID)) {
```

Comment:
The running tasks cannot be cancelled - the only way is to wait for them to terminate. Therefore I need to check if the test has been terminated in the meantime. It is also what we discussed.